src/Compilers/VisualBasic/Portable/Symbols/Tuples/TupleTypeSymbol.vb

```diff
@@ -0,0 +1,987 @@
+                        Dim tupleUnderlyingEvent As EventSymbol = (DirectCast(symbol, EventSymbol)).TupleUnderlyingEvent
+                        Dim associatedField As FieldSymbol = tupleUnderlyingEvent.AssociatedField
+
+                        If associatedField IsNot Nothing Then
+                            Debug.Assert(associatedField.ContainingSymbol Is Me._underlyingType)
+                            Debug.Assert(Me._underlyingType.GetMembers(associatedField.Name).IndexOf(associatedField) < 0)
+                            smallDictionary.Add(associatedField.OriginalDefinition, New TupleFieldSymbol(Me, associatedField, -i - 1))
+                        End If
+
+                        smallDictionary.Add(tupleUnderlyingEvent.OriginalDefinition, symbol)
+
+                    Case SymbolKind.Field
+                        Dim tupleUnderlyingField As FieldSymbol = (DirectCast(symbol, FieldSymbol)).TupleUnderlyingField
+                        If tupleUnderlyingField IsNot Nothing Then
+                            smallDictionary(tupleUnderlyingField.OriginalDefinition) = symbol
+                        End If
+
+                    Case SymbolKind.Method
+                        smallDictionary.Add((DirectCast(symbol, MethodSymbol)).TupleUnderlyingMethod.OriginalDefinition, symbol)
+
+                    Case SymbolKind.Property
+                        smallDictionary.Add((DirectCast(symbol, PropertySymbol)).TupleUnderlyingProperty.OriginalDefinition, symbol)
+
+                    Case Else
+                        Throw ExceptionUtilities.UnexpectedValue(symbol.Kind)
+                End Select
+
+                i -= 1
+            End While
+
+            Return smallDictionary
+        End Function
+
+        Public Function GetTupleMemberSymbolForUnderlyingMember(Of TMember As Symbol)(underlyingMemberOpt As TMember) As TMember
+            Dim result As TMember
+            If underlyingMemberOpt Is Nothing Then
+                result = Nothing
+            Else
+                Dim originalDefinition As Symbol = underlyingMemberOpt.OriginalDefinition
+                If originalDefinition.ContainingType Is Me._underlyingType.OriginalDefinition Then
+                    Dim symbol As Symbol = Nothing
+
+                    If Me.UnderlyingDefinitionToMemberMap.TryGetValue(originalDefinition, symbol) Then
+                        result = CType((CObj(symbol)), TMember)
+                        Return result
+                    End If
+                End If
+                result = Nothing
+            End If
+            Return result
+        End Function
+
+        Public Overrides Function GetMembers(name As String) As ImmutableArray(Of Symbol)
+            Return Me.GetMembers().WhereAsArray(Function(m As Symbol) m.Name = name)
+        End Function
+
+        Public Overrides Function GetTypeMembers() As ImmutableArray(Of NamedTypeSymbol)
+            ' do Not support nested types at the moment
```

Comment:
done

---
In reply to: [71571382](https://github.com/dotnet/roslyn/pull/12596#discussion_r71571382) [](ancestors = 71571382)